agent/util-scripts/pbench-tool-meister-start

```diff
@@ -244,13 +204,635 @@ def waitpid(pid):
+            # check is necessary because sometimes callers might have already
+            # specified a name that maps to 127.0.0.1, and Redis will throw an
+            # error if multiple names mapped to the same address.
+            try:
+                localhost_ip = socket.gethostbyname("localhost")
+            except socket.error:
+                # Interesting networking environment, no IP address for
+                # "localhost".  Just use the host we already have.
+                self.local_host = self.host
+            else:
+                if bind_host_ip != localhost_ip:
+                    assert (
+                        self.bind_host != "localhost"
+                    ), f"Logic Bomb!  self.bind_host ({self.bind_host:r}) == 'localhost'?"
+                    # The bind host name is not the same as "localhost" so we
+                    # can add it to the list of host names the Redis server
+                    # will bind to.
+                    bind_hostnames_l.append("localhost")
+                    self.local_host = "localhost"
+                else:
+                    # Whatever the self.bind_host is, it maps to the same IP
+                    # address as localhost, so just use the self.host for any
+                    # "local" access.
+                    self.local_host = self.host
+
+        bind_host_names = " ".join(bind_hostnames_l)
+
+        # Create the Redis server pbench-specific configuration file
+        redis_conf = tm_dir / "redis.conf"
+        params = {
+            "bind_host_names": bind_host_names,
+            "tm_dir": tm_dir,
+            "redis_port": self.bind_port,
+        }
+        try:
+            with redis_conf.open("w") as fp:
+                fp.write(self.conf_tmpl.format(**params))
+        except Exception as exc:
+            raise self.Err(
+                "failed to create redis server configuration", ReturnCode.EXCREDISCONFIG
+            ) from exc
+
+        # Start the Redis Server itself
+        redis_srvr = "redis-server"
+        redis_srvr_path = find_executable(redis_srvr)
+        self.pid_file = tm_dir / f"redis_{self.bind_port:d}.pid"
+        try:
+            retcode = os.spawnl(os.P_WAIT, redis_srvr_path, redis_srvr, redis_conf)
+        except Exception as exc:
+            raise self.Err(
+                "failed to create redis server, daemonized", ReturnCode.EXCSPAWNREDIS
+            ) from exc
+        else:
+            if retcode != 0:
+                raise self.Err(
+                    f"failed to create redis server, daemonized; return code: {retcode:d}",
+                    ReturnCode.REDISFAILED,
+                )
+
+    def kill(self, ret_val):
+        """kill - attempt to KILL the running Redis server.
+
+        This method is a no-op if the server instance isn't managed by us.
+
+        Returns ReturnCode "enum" via the "kill" return code method.
+        """
+        if self._managed is False:
+            return ret_val
```

Comment:
It seems like this should be calling `self.managed()` instead of referencing the attribute directly -- doing so would give you the chance to address the `self._managed is None` case without disturbing the code, here.

Reply:
Reworked.